src/grpc_server.cc

```diff
@@ -1340,6 +1340,226 @@ CommonHandler::SetUpAllRequests()
       inference::TraceSettingRequest, inference::TraceSettingResponse>(
       "Trace", 0, OnRegisterTrace, OnExecuteTrace, false /* async */, cq_);
 
+  //
+  //  Log Settings
+  //
+
+  auto OnRegisterLogging =
+      [this](
+          grpc::ServerContext* ctx, inference::LogSettingsRequest* request,
+          grpc::ServerAsyncResponseWriter<inference::LogSettingsResponse>*
+              responder,
+          void* tag) {
+        this->service_->RequestLogSettings(
+            ctx, request, responder, this->cq_, this->cq_, tag);
+      };
+
+  auto OnExecuteLogging = [this](
+                              inference::LogSettingsRequest& request,
+                              inference::LogSettingsResponse* response,
+                              grpc::Status* status) {
+
+#ifdef TRITON_ENABLE_LOGGING
+    TRITONSERVER_Error* err = nullptr;
+    // Update log settings
+    // Server and Core repos do not have the same Logger object
+    // Each update must be applied to both server and core repo versions
+    if (!request.settings().empty()) {
+      {
+        static std::string setting_name = "log_file";
+        auto it = request.settings().find(setting_name);
+        if (it != request.settings().end()) {
+          const auto& log_param = it->second;
+          if (log_param.parameter_choice_case() !=
+              inference::LogSettingsRequest_SettingValue::ParameterChoiceCase::
+                  kStringParam) {
+            err = TRITONSERVER_ErrorNew(
+                TRITONSERVER_ERROR_INVALID_ARG,
+                (std::string("expect string for '") + setting_name + "'")
+                    .c_str());
+            GOTO_IF_ERR(err, earlyexit);
+          } else {
+            // Set new settings in server then in core
+            const std::string& log_file_path = it->second.string_param();
+            bool success = LOG_SET_OUT_FILE(log_file_path);
+            // On failure, close log file and revert to default "empty"
+            if (!success) {
+              std::string empty;
+              TRITONSERVER_ServerOptionsSetLogFile(nullptr, empty.c_str());
```

Comment:
Don't change anything on error, reverting to previous state instead of back to default. i.e. if user have sucessfully set log file to "abc.log", later failure to set "def.log" shouldn't change the destination to standard error, otherwise more confusion is created.

Reply:
Previous state reversion now handled in logging.h, not here.